fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/AdvancedPaymentScheduleTransactionProcessor.java

```diff
@@ -809,7 +808,11 @@ protected void createNewTransaction(final LoanTransaction oldTransaction, final
             relations.add(LoanTransactionRelation.linkToTransaction(originalTransaction, newTransaction, CHARGEBACK));
         }
         loanChargeValidator.validateRepaymentTypeTransactionNotBeforeAChargeRefund(oldTransaction.getLoan(), oldTransaction, "reversed");
-        oldTransaction.reverse();
+
```

Comment:
THis is not needed. The issue was not with this logic!
Here there will be always "REPLAYED" relation, so checking it again is unnecessary!